### ISSUE TITLE
ALB Target Group missing name_prefix length check

### ISSUE THREAD

**Issue by shatil** [terraform-providers/terraform-provider-aws#1301]
tl;dr: `terraform plan` checks lengths for `name_prefix` field in `aws_alb`, but doesn't perform same validation for `aws_alb_target_group`. This results in a crash during `terraform apply` when `name_prefix` in latter resource exceeds six characters.

### Terraform Version
```
Terraform v0.9.11
```

### Affected Resource(s)
- `aws_alb_target_group`

### Terraform Configuration Files
```hcl
resource "aws_alb_target_group" "service" {
    name_prefix = "service-service"
    port = 80
    protocol = "HTTP"
    vpc_id = "vpc-c3f446fe"
}
```

### Expected Behavior
During `terraform plan` validation, I should receive the same error I do for resource `aws_alb`, like:

```
$ terraform plan
1 error(s) occurred:

* aws_alb.service: "name_prefix" cannot be longer than 6 characters: "service-service"
```

### Actual Behavior
`terraform plan` produced no validation error, but instead, `terraform apply` crashed because the ALB Target Group creation API doesn't accept names longer than 32 characters.

```
Error applying plan:

1 error(s) occurred:

* aws_alb_target_group.service: 1 error(s) occurred:

* aws_alb_target_group.service: Error creating ALB Target Group: ValidationError: Target group name 'service-service00e3a308cceb3fadadb87ab8de' cannot be longer than '32' characters
        status code: 400, request id: 9932faae-770b-11e7-8c77-bfd862635a94
```

### Steps to Reproduce
Use my above HCL snippet and:

1. `terraform plan`
2. `terraform apply`

### Important Factoids
Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? Custom version of OpenStack? Tight ACLs?

### References
Related to:

- https://github.com/hashicorp/terraform/issues/13995
- https://github.com/terraform-providers/terraform-provider-aws/issues/625

**Comment by riwiki** [comment]
This is no longer present in terraform 0.12.13.

**Comment by bflad** [comment]
Thanks for the nudge @riwiki! This issue was previously resolved in [version 1.7.0 of the Terraform AWS Provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#170-january-12-2018) and has been fixed in all versions since. 👍
